### PR TITLE
Reason-Phrase, reserved, unreserved, mark

### PR DESCRIPTION
**Pull request by KalitaAlexey**
Can parse "Call/Transaction Does Not Exist" now.

### CODE REVIEW

File: src/core/register_request_ext.rs [hunk 1]
```diff
@@ -11,6 +11,7 @@ pub trait RegisterRequestExt {
 
 impl RegisterRequestExt for SipMessage {
     fn expires(&self) -> Option<u32> {
-        self.contact_header_expires().or(self.expires_header())
+        self.contact_header_expires()
+            .or_else(|| self.expires_header())
```
Comment (KalitaAlexey) [inline src/core/register_request_ext.rs:15]: Clippy suggested that.

**Comment by bytebuddha** [comment]
Thanks Looks Great.

**Comment by bytebuddha** [comment]
@KalitaAlexey Hey I'm on vacation until the 22. I Will try to check github every now and then but i wont be able to really merge pull requests or answer questions until then

**Comment by KalitaAlexey** [comment]
You didn't merge it, did you? :-)

**Pull request merged**